### PR TITLE
Truncate hostname longer than 64 bits in Servicecomb registration

### PR DESCRIPTION
**Pull request by luanwenfei-venus**
【Fix issue】#1329

[Modification content] Truncate the hostname when registering Servicecomb and limit it to no more than 64 characters. Refer to spring-cloud-huawei's method https://github.com/huaweicloud/spring-cloud-huawei/pull/1072/files

[Use case description] Not involved

[Self-test situation] Local static check passed

[Scope of influence] No impact on users

### CODE REVIEW

File: sermant-plugins/sermant-service-registry/spring-cloud-registry-service/src/main/java/com/huawei/registry/service/client/ScClient.java [hunk 1]
```diff
@@ -407,18 +406,21 @@ private void initScClientAndWatch() {
...
     }
 
     private void buildMicroServiceInstance() {
         microserviceInstance = new MicroserviceInstance();
         microserviceInstance.setStatus(MicroserviceInstanceStatus.UP);
-        microserviceInstance.setHostName(RegisterContext.INSTANCE.getClientInfo().getHost());
+        String hostName = RegisterContext.INSTANCE.getClientInfo().getHost();
+        microserviceInstance.setHostName(
+                hostName != null && hostName.length() > MAX_HOST_NAME_LENGTH ? hostName.substring(0,
+                        MAX_HOST_NAME_LENGTH) : hostName);
```
Comment (lilai23) [inline sermant-plugins/sermant-service-registry/spring-cloud-registry-service/src/main/java/com/huawei/registry/service/client/ScClient.java:423]: RegisterContext.INSTANCE.getClientInfo().getHost()有多个地方调用，microserviceInstance的hostname和RegisterContext的不一致， 可能会有不匹配的情况？建议在 RegisterContext.INSTANCE.getClientInfo().setHost方法里面做长度截断

File: sermant-plugins/sermant-service-registry/dubbo-registry-service/src/main/java/com/huawei/dubbo/registry/service/RegistryServiceImpl.java [hunk 2]
```diff
@@ -394,7 +396,11 @@ private void createMicroserviceInstance() {
 
     private String getHost() {
         try {
-            return InetAddress.getLocalHost().getHostName();
+            String hostName = InetAddress.getLocalHost().getHostName();
+
+            // kie不支持64个字符以上的hostname注册, 此处参考spring-cloud-huawei进行截断处理
```
Comment (provenceee) [inline sermant-plugins/sermant-service-registry/dubbo-registry-service/src/main/java/com/huawei/dubbo/registry/service/RegistryServiceImpl.java]: 不是kie，是Servicecomb

Comment (luanwenfei-venus) [reply sermant-plugins/sermant-service-registry/dubbo-registry-service/src/main/java/com/huawei/dubbo/registry/service/RegistryServiceImpl.java]: 已修改

File: sermant-plugins/sermant-service-registry/spring-cloud-registry-service/src/main/java/com/huawei/registry/service/client/ScClient.java (hunk 1 above)
Comment (luanwenfei-venus) [reply sermant-plugins/sermant-service-registry/spring-cloud-registry-service/src/main/java/com/huawei/registry/service/client/ScClient.java:423]: 先修改注册的hostname，只有servicecomb注册时进行截断

**Review by lilai23 — APPROVED** [review]

**Review by provenceee — APPROVED** [review]

**Pull request merged**